File: wantedMarketAPI/src/main/java/com/example/wantedmarketapi/domain/product/ProductInfo.java
package com.example.wantedmarketapi.domain.product;

import com.example.wantedmarketapi.domain.product.Product.Status;
import lombok.Getter;

@Getter
public class ProductInfo {
    private final Long id;
    private final String name;
    private final Integer price;
    private final Status status;
    private final Long userId;

    public ProductInfo(Product product) {
        this.id = product.getId();
        this.name = product.getName();
        this.price = product.getPrice();
        this.status = product.getStatus();
        this.userId = product.getUserId();
    }


}

File: api/src/main/kotlin/wanted/market/api/domain/product/dto/ProductInfo.kt
package wanted.market.api.domain.product.dto

import wanted.market.api.domain.product.entity.Product


data class ProductInfo(
    var id: Long,
    var name: String,
    var price: Int,
    var status: String,
    var sellerId: Long
) {
    companion object {
        fun from(product: Product): ProductInfo {
            return ProductInfo(
                id = product.id!!,
                sellerId = product.seller.id!!,
                name = product.name,
                price = product.price,
                status = product.status.value
            )
        }
    }
}
